src/aap_eda/core/utils/credentials.py

```diff
@@ -148,7 +192,128 @@ def validate_schema(schema: dict) -> list[str]:
             errors.append("required must be a list of strings")
         else:
             for field_id in required_fields:
-                if field_id not in field_ids:
+                if field_id not in id_fields:
                     errors.append(f"required field {field_id} does not exist")
 
     return errors
+
+
+def validate_injectors(schema: dict, injectors: dict) -> dict:
+    errors = []
+    context = _default_context(schema)
+
+    if not isinstance(injectors, dict):
+        errors.append(
+            "injectors must be a dict type and defines keys"
+            f" in {SUPPORT_KEYS_IN_INJECTORS}"
+        )
+
+    if not any(
+        support_key in injectors.keys()
+        for support_key in SUPPORT_KEYS_IN_INJECTORS
+    ):
+        errors.append(
+            "injectors must have keys defined in the"
```

Comment:
@hsong-rh Since each error is a separate line, we need to have injectors as Injectors.